src/main/java/uk/gov/hmcts/ccd/domain/model/definition/CaseEventDefinition.java

```diff
@@ -205,4 +214,29 @@ public void setCanSaveDraft(Boolean canSaveDraft) {
             .filter(f -> f.getCaseFieldId().equals(caseFieldId))
             .findFirst();
     }
+
+    public Map<String, JsonNode> buildJsonNodeFromCaseFieldsWithDefaultValue() {
+        Map<String, JsonNode> data = new HashMap<>();
+
+        getCaseFields().forEach(
+            caseField -> {
+
+                List<JsonNode> collect = caseField.getCaseEventFieldComplexDefinitions().stream()
+                    .filter(e -> e.getDefaultValue() != null)
+                    .filter(e -> !e.getReference().isBlank())
+                    .map(caseEventFieldComplex -> JacksonUtils.buildFromDottedPath(caseEventFieldComplex.getReference(),
+                                                                                   caseEventFieldComplex.getDefaultValue())).collect(toList());
+
+                ObjectNode objectNode = MAPPER.getNodeFactory().objectNode();
+                collect.forEach(e -> {
+                    String next = e.fieldNames().next();
+                    objectNode.set(next, e.findValue(next));
+                });
+                if (!collect.isEmpty()) { // to prevent construct like "FieldA": {}
+                    data.put(caseField.getCaseFieldId(), objectNode);
+                }
+            });
+
+        return data;
+    }
```

Comment:
mmm not sure whether it's the best thing in terms of separation of concerns on having a definition object having logic to build JSON...

Reply:
Not a problem. Can move it to the Service layer. Was just thinking of 'anemic domain model'

Reply:
Yes, Good to move this logic to service class